File: python/python_bandwidth_bxml/verbs/redirect.py
"""
redirect.py

Representation of Bandwidth's redirect BXML verb

@author Jacob Mulford
@copyright Bandwidth INC
"""

from lxml import etree

from .base_verb import AbstractBxmlVerb

REDIRECT_TAG = "Redirect"


class Redirect(AbstractBxmlVerb):

    def __init__(self, redirect_url=None, redirect_method=None, tag=None):
        """
        Initializes the Redirect class with the following parameters

        :param str redirect_url: The url to retrieve the next BXML
        :param str redirect_method: The HTTP method used to retrieve the next url
        :param str tag: Optional tag to include in the callback
        """
        self.redirect_url = redirect_url
        self.redirect_method = redirect_method
        self.tag = tag

    def to_xml(self):
        root = etree.Element(REDIRECT_TAG)
        if self.redirect_url is not None:
            root.set("redirectUrl", self.redirect_url)
        if self.redirect_method is not None:
            root.set("redirectMethod", self.redirect_method)
        if self.tag is not None:
            root.set("tag", self.tag)
        return etree.tostring(root).decode()

File: php/BxmlVerbs/verbs/redirect.php
<?php
/**
 * redirect.php
 *
 * Implementation of the BXML Redirect verb
 *
 * @author Jacob Mulford
 * @copyright Bandwidth INC
 */

namespace BxmlVerbs;

require_once "verb.php";

class Redirect extends Verb {

    /**
     * Sets the username attribute for PlayAudio
     *
     * @param string $username The username for http authentication on the audio url
     */
    public function username($username) {
        $this->username = $username;
    }

    /**
     * Sets the password attribute for PlayAudio
     *
     * @param string $password The password for http authentication on the audio url
     */
    public function password($password) {
        $this->password = $password;
    }

    /**
     * Sets the redirectUrl attribute for PlayAudio
     *
     * @param string $redirectUrl The url to receive the redirect callback 
     */
    public function redirectUrl($redirectUrl) {
        $this->redirectUrl = $redirectUrl;
    }

    /**
     * Sets the redirectMethod attribute for PlayAudio
     *
     * @param string $redirectMethod The http method to send the redirect callback 
     */
    public function redirectMethod($redirectMethod) {
        $this->redirectMethod = $redirectMethod;
    }

    /**
     * Sets the tag attribute for PlayAudio
     *
     * @param string $tag A custom string to be included in callbacks 
     */
    public function tag($tag) {
        $this->tag = $tag;
    }

    public function toXml($doc) {
        $element = $doc->createElement("Redirect");

        if(isset($this->username)) {
            $element->setAttribute("username", $this->username);
        }

        if(isset($this->password)) {
            $element->setAttribute("password", $this->password);
        }

        if(isset($this->tag)) {
            $element->setAttribute("tag", $this->tag);
        }

        if(isset($this->redirectUrl)) {
            $element->setAttribute("redirectUrl", $this->redirectUrl);
        }

        if(isset($this->redirectMethod)) {
            $element->setAttribute("redirectMethod", $this->redirectMethod);
        }

        return $element;
    }
}
